ast/literal.go

```diff
@@ -12,6 +13,39 @@ type LiteralNode struct {
 	Posx  Pos
 }
 
+// NewLiteralNode returns a new literal node representing the given
+// literal Go value, which must correspond to one of the primitive types
+// supported by HIL. Lists and maps cannot currently be constructed via
+// this function.
+//
+// If an inappropriately-typed value is provided, this function will
+// panic. The main intended use of this function is to produce "synthetic"
+// literals from constants in code, where the value type is well known
+// at compile time.
+func NewLiteralNode(value interface{}, pos Pos) *LiteralNode {
+	goType := reflect.TypeOf(value)
+	var hilType Type
+
+	switch goType.Kind() {
+	case reflect.Bool:
+		hilType = TypeBool
+	case reflect.Int:
+		hilType = TypeInt
+	case reflect.Float64:
+		hilType = TypeFloat
+	case reflect.String:
+		hilType = TypeString
+	default:
+		panic(fmt.Errorf("unsupported literal node type: %T", value))
```

Comment:
I'm unsure if we should panic here. I think it makes more sense to return an error since I can imagine cases where you may want dynamic values coming in here (that we can't be sure won't panic).